home/components/ProfileUnauthenticated.tsx

```diff
@@ -1,19 +1,82 @@
-import { useNavigation } from '@react-navigation/native';
+import * as WebBrowser from 'expo-web-browser';
 import * as React from 'react';
 import { Platform, StyleSheet, TouchableOpacity, View } from 'react-native';
+import url from 'url';
 
+import Analytics from '../api/Analytics';
+import ApolloClient from '../api/ApolloClient';
+import Config from '../api/Config';
+import Colors from '../constants/Colors';
+import { useDispatch } from '../redux/Hooks';
+import SessionActions from '../redux/SessionActions';
 import ScrollView from './NavigationScrollView';
 import PrimaryButton from './PrimaryButton';
 import { StyledText } from './Text';
 
 export default function ProfileUnauthenticated() {
-  const navigation = useNavigation();
-  const _handleSignInPress = () => {
-    navigation.navigate('SignIn');
+  const dispatch = useDispatch();
+  const [isAuthenticating, setIsAuthenticating] = React.useState(false);
+  const [authenticationError, setAuthenticationError] = React.useState<string | null>(null);
+  const mounted = React.useRef<boolean | null>(true);
+
+  React.useEffect(() => {
+    mounted.current = true;
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
+
+  const _handleSignInPress = async () => {
+    await _handleAuthentication('login', Analytics.events.USER_LOGGED_IN);
+  };
+
+  const _handleSignUpPress = async () => {
+    await _handleAuthentication('signup', Analytics.events.USER_CREATED_ACCOUNT);
   };
 
-  const _handleSignUpPress = () => {
-    navigation.navigate('SignUp');
+  const _handleAuthentication = async (urlPath: string, analyticsEvent: string) => {
+    if (isAuthenticating) {
+      return;
+    }
+    setAuthenticationError(null);
+    setIsAuthenticating(true);
+
+    try {
+      const redirectBase = 'expauth://auth';
+      const authSessionURL = `${
+        Config.website.origin
+      }/${urlPath}?app_redirect_uri=${encodeURIComponent(redirectBase)}`;
+      const result = await WebBrowser.openAuthSessionAsync(authSessionURL, redirectBase);
+
+      if (!mounted.current) {
+        return;
+      }
+
+      if (result.type === 'success') {
+        const resultURL = url.parse(result.url, true);
+        const sessionSecret = resultURL.query['session_secret'] as string;
+        const usernameOrEmail = resultURL.query['username_or_email'] as string;
+
```

Comment:
should we add some assertions here for `sessionSecret` and `usernameOrEmail`?

```
assert(sessionSecret, `session_secret is blank in auth redirect query`);
assert(usernameOrEmail, `username_or_email is blank in auth redirect query`);
```

or perhaps something more useful for presentation for users - but something that can help us quickly identify what has gone wrong for them and that we can put into sentry (when we actually use it on home)